📖 Remove outdated capi book roadmap (#6680)
kubernetes-sigs/cluster-api: docs/book/src/roadmap.md

# Cluster API Roadmap

> The roadmap is a constant work in progress, subject to frequent revision. You can find it at [GitHub](https://github.com/kubernetes-sigs/cluster-api/discussions). Please feel free to participate in the discussion!

Reviewer: Might be better to link directly to the specific discussion

```suggestion
You can find the Cluster API roadmap discussion at [GitHub](https://github.com/kubernetes-sigs/cluster-api/discussions/5556). Please feel free to participate!
```

Author: Sounds like a solid suggestion. Thanks!
